selendroid-common/src/main/java/io/selendroid/common/SelendroidCapabilities.java

```diff
@@ -395,17 +395,13 @@ private String getDefaultVersion(Set<String> keys, String appName) {
     return listOfApps.size() > 0 ? listOfApps.last() : null;
   }
 
-  // throws exception if user didn't pass the capability as a boolean
+  // throws exception if user didn't pass the capability as a boolean or String parsable as boolean
   private Boolean getBooleanCapability(String key) {
     Object o = getRawCapabilities().get(key);
     if (o == null) {
       return null;
-    } else if (o instanceof Boolean) {
-      return (Boolean) o;
     } else {
-      throw new ClassCastException(String.format(
-          "DesiredCapability %s's value should be boolean: found value %s of type %s", key, o.toString(), o.getClass()
-              .getName()));
+      return is(key);
```

Comment:
problem is, i'm still here to listen to user complaints / issues later on. And I much prefer error messages with explicit strings attached than something that essentially is a side-effect.

So, I'll merge it if you change to checking explicitly for 'true' or 'false' strings in addition to the previous Boolean instance check, but I don't want the is() behavior that selenium core library provides as it doesn't correct users doing things poorly (and if we did actually use the is(), we'd need some documentation somewhere to explain what the side effects are... otherwise, more confused users, more noisy issues).

Add on top to the fact that this change is only getting in place because some selenium library doesn't allow you to pass actual boolean values as capabilities (which is a bug in that library :) ).